modules/generator/registry/histogram.go

```diff
@@ -171,16 +187,27 @@ func (h *histogram) collectMetrics(appender storage.Appender, timeMs int64, exte
 			lb.Set(name, s.labels.values[i])
 		}
 
+		// If we are about to call Append for the first time on a series,
+		// we need to first insert a 0 value to allow Prometheus to start from a non-null value.
+		if s.isNew() {
+			lb.Set(labels.MetricName, h.nameCount)
```

Comment:
Question, maybe it makes sense to create a new ìnit` method in the interface:

```go
type metric interface {
	name() string
        initMetrics(appender storage.Appender,...)
	collectMetrics(appender storage.Appender, timeMs int64, externalLabels map[string]string) (activeSeries int, err error)
	removeStaleSeries(staleTimeMs int64)
}
```
since I see we do the same logic in the counter

Reply:
This logic is actually of the series, not the metric. It needs passing a couple of values, the function would actually encapsulate very little. I don't think it's worth the hassle.

```go
func (co *counterSeries) init(appender storage.Appender, lbls labels.Labels, ts int64) {
	_, _ = appender.Append(0, lbls, ts, 0)
	co.registerSeenSeries()
}
```